File: frontend/src/types/scout.ts
export interface TalentSummary {
  id: string;
  desired_job: string | null;
  desired_region: string | null;
  is_experienced: boolean;
  license_types: string[];
  career_count: number;
  summary_preview: string | null;
  created_at: string;
  updated_at: string;
}

export interface TalentListResponse {
  items: TalentSummary[];
  page: number;
  size: number;
  total: number;
}

export interface ScoutRead {
  id: string;
  company_id: string;
  company_name: string | null;
  user_id: string;
  job_post_id: string | null;
  job_title: string | null;
  status: string;
  message: string | null;
  created_at: string;
  updated_at: string;
}

export interface ScoutListResponse {
  items: ScoutRead[];
  total: number;
}

export interface ScoutCreateInput {
  resume_id: string;
  job_post_id?: string;
  message?: string;
}

File: backend/app/schemas/scout.py
from datetime import datetime

from pydantic import BaseModel, Field


# --- Talent Search (anonymized resume for companies) ---


class TalentSummary(BaseModel):
    id: str
    desired_job: str | None = None
    desired_region: str | None = None
    is_experienced: bool = False
    license_types: list[str] = []
    career_count: int = 0
    summary_preview: str | None = None
    created_at: datetime
    updated_at: datetime


class TalentListResponse(BaseModel):
    items: list[TalentSummary]
    page: int
    size: int
    total: int


# --- Scout ---


class ScoutCreate(BaseModel):
    resume_id: str = Field(..., description="대상 이력서 ID")
    job_post_id: str | None = Field(None, description="연결할 공고 ID (선택)")
    message: str | None = Field(None, max_length=2000, description="스카우트 메시지")


class ScoutRead(BaseModel):
    id: str
    company_id: str
    company_name: str | None = None
    user_id: str
    job_post_id: str | None = None
    job_title: str | None = None
    status: str
    message: str | None = None
    created_at: datetime
    updated_at: datetime


class ScoutListResponse(BaseModel):
    items: list[ScoutRead]
    total: int


class ScoutRespondRequest(BaseModel):
    status: str = Field(
        ...,
        pattern="^(ACCEPTED|REJECTED|HOLD)$",
        description="응답 상태",
    )
